internal/cmd/temporal-features-test-proxy/main.go

```diff
@@ -0,0 +1,752 @@
+package main
```

Comment:
It's not immediately obvious to me why this guy needs to be his own executable? Seemingly the runner could start this up internally rather than needing to start it as a subprocess?

Reply:
Concur. @chronos-tachyon - take a peek at what I recently had to do with HTTP proxying: https://github.com/temporalio/features/pull/448 (ignore the subprocess stuff which was because I had to test environment variables safely and unrelated to the HTTP proxy part). Some notes:

* Can have a feature's config opt-in to the proxy so if someone runs a specific feature without needing a proxy it's not started
* Can name it "tcp proxy" to disambiguate from http proxy (note, I called the other "http proxy" to avoid this confusion too)
* Can make a `TCPProxy` utility as a Go library that the command just happens to use. This can help properly encapsulate the tool (even though it'll only ever be used by the cmd) instead of using globals and subprocesses and such
* Features know if they need a proxy, so they can choose to, say, dial to that address instead of the default one. No need to affect existing running features by making them go through the proxy.

Reply:
I couldn't do it in-process because, when I did, there were too many CPU-bound goroutines to allow the proxy server's goroutines to get serviced in a timely manner (delays on the order of 20 seconds).